### PR TITLE
Amaps as library

### PR DESCRIPTION
**Pull request by ghost**
An amaps library added, which gets built with new npm scripts. Naming of npm scripts can be discussed, or altered later.

the amaps library (src/index.js) contains code required to format and parse requests/responses from Amsterdam's BAG API. This includes encapsulating the coordinate transformation with proj4.

### CODE REVIEW

**Comment by ghost** [comment]
Note that circleci build will fail. Builds/tests work with docker-compose.

**Comment by ghost** [comment]
Oh, I see I need to update the documentation yet.

**Comment by ghost** [comment]
There will also be a few things to iron out when @boris-arkenaar's work with the docker-based build setup is merged. This pull request adds a few new docker containers and npm scripts, but there are also a few changes to the nodejs scripts for building and testing which might need to be deconflicted.

File: package.json [hunk 1]
```diff
@@ -22,14 +25,23 @@
...
+    "shx": "^0.2.2",
     "tap": "^11.1.3"
   },
   "scripts": {
     "test": "node scripts/serve",
     "watch-test": "nodemon --watch scripts --watch test -e js,html scripts/serve.js",
-    "lint": "eslint test scripts",
-    "serve": "http-server test"
+    "serve": "http-server test",
+    "lint": "eslint test scripts && eslint -c .eslintrc-amaps.js src",
```
Comment (boris-arkenaar) [inline package.json]: I think you could move `.eslintrc-amaps.js` to `src/.eslintrc.js` and then change this line into: `"lint": "eslint test scripts src",`.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+//user-provided function for featureQuery to parse Ajax response
+function responseFormatter(res) {
+  let filtered = res.results.filter(x => x.hoofdadres === true);
+  return filtered.length > 0 ? filtered[0] : null;
+}
+const amaps = {
+  BagApiRequestFormatter: requestFormatter,
+  BagApiResponseFormatter: responseFormatter
```
Comment (boris-arkenaar) [inline src/index.js]: Why do you start these properties upper case?

**Review by boris-arkenaar — COMMENTED** [review]
The iife files should not be added to the repo I guess. At least no the ones in the `test` disrectory.
Besides a few remarks, looks great!

File: src/index.js (hunk 2 above)
Comment (ghost) [reply src/index.js]: to indicate Bag was a proper noun ... but I've changed it to be proper camelCase.

File: package.json (hunk 1 above)
Comment (ghost) [reply package.json]: Good one! done.

**Comment by ghost** [comment]
@boris-arkenaar I've addressed the things you mentioned and also expanded the documentation to describe both nlmaps build and amaps build.

**Review by boris-arkenaar — APPROVED** [review]

**Pull request merged**